src/utils/Processing.ts

```diff
@@ -317,45 +315,33 @@ export async function loadExternalDependencies(
 ): Promise<void> {
   // Add FHIR to the dependencies so it is loaded
   const dependencies = (config.dependencies ?? []).slice(); // slice so we don't modify actual config;
-  const fhirVersion = config.fhirVersion.find(v => isSupportedFHIRVersion(v));
-  let fhirPackageId: string;
-  let prerelease = false;
-  if (/^4\.0\./.test(fhirVersion)) {
-    fhirPackageId = 'hl7.fhir.r4.core';
-  } else if (/^(4\.1\.|4\.3.\d+-)/.test(fhirVersion)) {
-    fhirPackageId = 'hl7.fhir.r4b.core';
-    prerelease = true;
-  } else if (/^4\.3.\d+$/.test(fhirVersion)) {
-    fhirPackageId = 'hl7.fhir.r4b.core';
-  } else if (/^5\.0.\d+$/.test(fhirVersion)) {
-    fhirPackageId = 'hl7.fhir.r5.core';
-  } else {
-    fhirPackageId = 'hl7.fhir.r5.core';
-    prerelease = true;
-  }
-  if (prerelease) {
+  const fhirVersionInfo = config.fhirVersion
+    .map(v => getFHIRVersionInfo(v))
+    .find(v => v.isSupported);
+  if (fhirVersionInfo.isPreRelease) {
```

Comment:
Ahhhhh yeah. That's the good stuff right there. Thank you for `getFhirVersionInfo`.